File: MatricesJava/src/Ejercicio7.java
package paquete;

import java.util.Scanner;

public class Ejercicio7 {

    public static void main(String[] args) {
        var scanner = new Scanner(System.in);
        int cantidadFilas, cantidadColumnas, contador = 0;
        System.out.print("Ingrese la cantidad de filas: ");
        cantidadFilas = scanner.nextInt();
        System.out.print("Ingrese la cantidad de columnas: ");
        cantidadColumnas = scanner.nextInt();
        float matriz[][] = new float[cantidadFilas][cantidadColumnas];
        int tamanoVector = cantidadFilas * cantidadColumnas;
        float vector[] = new float[tamanoVector];
        for (int i = 0; i < cantidadFilas; i++) {
            for (int j = 0; j < cantidadColumnas; j++) {
                System.out.print("Ingrese un numero para la posicion: ("
                        + (i + 1) + "-" + (j + 1) + "): ");
                matriz[i][j] = scanner.nextInt();
                vector[contador++] = matriz[i][j];
            }
        }
        for (int i = 0; i < tamanoVector; i++) {
            System.out.println("Posicion : " + (i + 1) + " numero: " + vector[i]);
        }
    }
}

File: MatricesC++/Ejercicio7.cpp
#include<iostream>
#include<conio.h>

using namespace std;

int main(){
	/*Hacer un programa que llene una matriz de 10*10 y que almacene en la diagonal principal unos y en las demas posiciones ceros.*/
	int cantidadFilas  = 0, cantidadColumnas = 0, contadorVector = 0;
	int matriz[10][10],  vector[cantidadFilas * cantidadColumnas];
	cout<<"Por favor ingrese la cantiadd de filas: "; cin >> cantidadFilas; 
	cout<<"Por favor ingrese la cantidad de columnas: "; cin >> cantidadColumnas;
	for(int i = 0; i < cantidadFilas; i++){
		for(int j = 0; j < cantidadColumnas; j++){
			cout<<"Por favor ingrese un numero para la posicion: ("<<i+1<<"-"<<j+1<<"): "; cin >> matriz[i][j];
			vector[contadorVector] = matriz[i][j];
			contadorVector++;
		}
	}	
	for(int i = 0; i < cantidadColumnas * cantidadFilas; i++){
		cout<<"Elemento: ("<<i+1<<"): "<< vector[i] <<endl;
	}

	getch();
	return 0; 
}
